app/models/date_attended.rb

```diff
@@ -12,9 +12,19 @@
 
 class DateAttended < ActiveRecord::Base
   belongs_to :fee
+  belongs_to :expense
 
   validates :date, presence: true
-  validates :fee, presence:  true
+  validates :fee, presence:  true, if: "expense.nil?"
+  validates :expense, presence:  true, if: "fee.nil?"
+  validate  :belongs_to_fee_or_expense
+
+  def belongs_to_fee_or_expense
```

Comment:
Having a polymorphic relationship between fees, expenses and dates attended will obviate the need for this validation.